src/OmniSharp.Host/Startup.cs

```diff
@@ -189,15 +189,28 @@ public void ConfigureServices(IServiceCollection services)
                 }
                 catch (Exception e)
                 {
-                    var message = $"The project system '{projectSystem.GetType().Name}' threw exception during initialization.\n{e.Message}\n{e.StackTrace}";
+                    var message = $"The project system '{projectSystem.GetType().Name}' threw exception during initialization.";
                     // if a project system throws an unhandled exception it should not crash the entire server
-                    logger.LogError(message);
+                    logger.LogError(e, message);
+                }
+            }
+
+            // run all workspace options providers
+            foreach (var workspaceOptionsProvider in PluginHost.GetExports<IWorkspaceOptionsProvider>())
+            {
+                try
+                {
+                    Workspace.Options = workspaceOptionsProvider.Process(Workspace.Options);
```

Comment:
I wanted to avoid getting into that yet, because that would need a more generic solution anyway - for the plugins.
I believe the best approach would be to use MEF metadata something like: 

```
    public interface IPlugin
    {
        string Name { get; }
        int Order { get; }
    }

    [MetadataAttribute]
    public class PluginMetadataAttribute : ExportAttribute, IPluginMetadata
    {
        public string Name { get; set; }
        public int Order { get; set; }

        public PluginMetadataAttribute()
            : base(typeof(IPlugin))
        {
        }
    }

[PluginMetadata(Name = "DefaultCSharpFormatting", Order = -1)]
```

And then use this metadata when plugins are discovered; this would apply to all plugins in the future. However, I believe we would need to design it a bit more carefully - I'll open a separate issue to track that